test/integration/formatting_stack/strategies/impl.clj

```diff
@@ -33,3 +34,13 @@
     "src/../src/formatting_stack/strategies/impl.clj")
 
   (is (spec-assertion-thrown? ::sut/existing-files (sut/absolutize "git" ["I_dont_exist"]))))
+
+(deftest git-ref-exists?
+  (are [input expected] (testing input
+                          (is (= expected
+                                 (sut/git-ref-exists? input)))
```

Comment:
Is the `testing` necessary here? Using `is` inside `are` causes superfluous assertions. (sometimes _definitely_  worth it btw.)

Butreporting is fine without `testing`, because the input is already shown clearly;
```clj
(clojure.test/is ((constantly false) "My input"))

FAIL in () (form-init10638740795635241686.clj:1)
expected: ((constantly false) "My input")
  actual: false
=> false
```


wdyt?

Reply:
I find it a good habit, given the following considerations:

* I prefer looking at the `testing`  strings vs. the "inlined code replacements" (for lack of a better name) that clojure.test performs
  * looking at the what (vs. the how) tends to suit better my style of thinking
* having a `testing` readily available is _designed for growth_
  * e.g. more args may be added tomorrow
    * if there wasn't a `testing` already there, one can be lazier about introducing a `testing` (and having a noisier diff later)
* `testing` forms part of the snippet I use and foster ([1](https://raw.githubusercontent.com/zenmacs/.emacs.d/fbed5d4b5e44ad65282c10a1b647d4fa014765e6/snippets/clojure-mode/are), [2](https://raw.githubusercontent.com/zenmacs/.emacs.d/fbed5d4b5e44ad65282c10a1b647d4fa014765e6/snippets/clojure-mode/dft))
  * `are` is admittledly a somewhat delicate construct.
  * by using a snippet, one can use `are` without having to think about intricacies
    * having to remember to add/remove `testing` breaks that 'abstraction'

Perhaps someday we'll have a higher-level `are` that is flawless, meanwhile I favor using it in an homogeneous manner that doesn't make people think too much :)

Reply:
Your first and last considerations boil down to personal preferences. The second consideration is valid, but should be considered on a case to case basis. Given the triviality of this unit test I don't think it needs a more extensive `are`.

However; this part of the diff doesn't warrant a big discussion either.